### ISSUE TITLE
rosbag recorder triggerd by arming

### ISSUE THREAD

**Issue by vooon** [mavlink/mavros#385]
I think it might be useful to start log recording on FCU arming.

**Comment by mhkabir** [comment]
I like it :dancer:
Ability to call custom rosbag record script would be nice as well. Till now, I had been manually doing this, but automation would be a plus.

**Comment by vooon** [comment]
I'm actually think to make custom `record` script. As i know it is python-based, so i can base on it.
Alternatively we may to run program, in that case we may automate "do at arming event" anything.

**Comment by meuchel** [comment]
Because rosbag is anyway a standalone program I like more the generic one to be able to run any program.
If already on track with a module for event triggered actions you should also think about other events which could trigger something. E.g. disarm or others.

**Comment by vooon** [comment]
I checked rosbag and found that `play` and `record` is native binaries.
So generic only. Next thing is on what stack to write: python or c++.

Proposed config:
```yaml
some_node: # only used in log
  event: armed # not sure that it needed, for now i only want armning
  action: run, stop # required, start on run, stop on disarm
  name: rosbag # optional, define node name
  restart: true # optional, restart? not sure that it'll be implemented soon
  output: log # optional, same as <node output="" />
  rosrun: rosbag record -a # like rosrun
some_script_1:
  event: armed
  shell: turn_off_wifi.sh # any executable, 
some_script_2:
  event: disarmed
  shell: turn_on_wifi.sh
```

**Comment by meuchel** [comment]
Looks good.
If there is rosrun there should be also roslaunch possible?

What about during same session if armed then disarmed and  armed again?
Then e.g. script will be started again and user must take care of double start? (afaik rosrun handles this itself)
In this cases event: disarmed would make sense.

**Comment by vooon** [comment]
Perhaps roslaunch is too complex.

Yes, rosrun kills previous node with same name. That don't work for anonymous nodes.

Also first must be more verbose:
```yaml
event: armed, disarmed
action: run, stop
```

**Comment by vooon** [comment]
http://wiki.ros.org/roslaunch/API%20Usage - helps a lot. Last thing - good name for that node.

eventlauncher?

**Comment by meuchel** [comment]
of course also integration of roslaunch would be nice. 
Although all could be handled already only with the script option (including rosrun).
Regarding the name here some ideas:
event_handler_node
event_launcher_node
mavros_events_node

**Comment by vooon** [comment]
Shell-launcher complete. Unfortunate stdout/stderr logging do not work (break execution).

Example config: [event_launcher.yaml](https://github.com/mavlink/mavros/blob/c368f38417c9b9156cb457a2124a4f2a21eb8555/mavros/launch/event_launcher.yaml)

As you can see there was ability to define custom trigger for event.
```yaml
event_name: # used in `event`
  service: service_path # std_srvs/Trigger
```

Next thing is rosrun.

**Comment by vooon** [comment]
Hmm, implementing rosrun has problems.

I think that we may use shell nodes for all our needs.

**Comment by vooon** [comment]
Closed by #386.

Shell entries just works fine.
